pwa/app/lib/utils.ts

```diff
@@ -8,3 +9,58 @@ export function cn(...inputs: ClassValue[]) {
 export function parseDateString(date: string) {
   return new Date(date);
 }
+
+export function sortMatchComparator(a: Match, b: Match) {
+  const compLevelValues: Record<string, number> = {
+    f: 5,
+    sf: 4,
+    qf: 3,
+    ef: 2,
+    qm: 1,
+  };
+  if (a.comp_level !== b.comp_level) {
+    return (
+      (compLevelValues[a.comp_level] ?? 0) -
+      (compLevelValues[b.comp_level] ?? 0)
+    );
+  }
+
+  return a.set_number - b.set_number || a.match_number - b.match_number;
+}
+
+export function sortMatches(matches: Match[]) {
+  return matches.sort(sortMatchComparator);
+}
+
+export function timestampsAreOnDifferentDays(
```

Comment:
This might break since timestamps are in UTC, not the event's timezone.

Reply:
at the time of writing, timezones were a future me problem

Reply:
I put in a todo at the caller location in the component for now